packages/react-paypal-js/src/v6/hooks/useVenmoOneTimePaymentSession.test.ts

```diff
@@ -0,0 +1,537 @@
+
+            expect(mockVenmoSession.destroy).not.toHaveBeenCalled();
+            expect(
+                mockSdkInstance.createVenmoOneTimePaymentSession,
+            ).not.toHaveBeenCalled();
+        });
+    });
+
+    describe("handleClick", () => {
+        test("should start session with presentation mode and orderId", async () => {
+            const props: UseVenmoOneTimePaymentSessionProps = {
+                presentationMode: "popup",
+                orderId: "test-order-id",
+                onApprove: expect.any(Function),
+                onCancel: expect.any(Function),
+                onError: expect.any(Function),
+            };
+
+            const { result } = renderHook(() =>
+                useVenmoOneTimePaymentSession(props),
+            );
+
+            await act(async () => {
+                await result.current.handleClick();
+            });
+
+            expect(mockVenmoSession.start).toHaveBeenCalledWith({
+                presentationMode: "popup",
+            });
+        });
+
+        test("should start session with createOrder when provided", async () => {
+            const mockCreateOrder = jest
+                .fn()
+                .mockReturnValue(Promise.resolve("created-order-id"));
+
+            const props: UseVenmoOneTimePaymentSessionProps = {
+                presentationMode: "popup",
+                createOrder: mockCreateOrder,
+                onApprove: expect.any(Function),
+                onCancel: expect.any(Function),
+                onError: expect.any(Function),
+            };
+
+            const { result } = renderHook(() =>
+                useVenmoOneTimePaymentSession(props),
+            );
+
+            await act(async () => {
+                await result.current.handleClick();
+            });
+
+            // The hook calls createOrder() and passes the result as second argument to start()
+            expect(mockVenmoSession.start).toHaveBeenCalledWith(
+                { presentationMode: "popup" },
+                expect.any(Promise), // createOrder() returns a Promise
+            );
+            expect(mockCreateOrder).toHaveBeenCalled();
+        });
+
+        test("should throw error when session is not available", async () => {
+            const props: UseVenmoOneTimePaymentSessionProps = {
+                presentationMode: "popup",
+                orderId: "test-order-id",
+                onApprove: expect.any(Function),
+                onCancel: expect.any(Function),
+                onError: expect.any(Function),
+            };
+
+            const { result, unmount } = renderHook(() =>
+                useVenmoOneTimePaymentSession(props),
+            );
+
+            // Unmount to destroy session
+            unmount();
+
+            await expect(
+                act(async () => {
+                    await result.current.handleClick();
+                }),
+            ).rejects.toThrow("Venmo session not available");
```

Comment:
```suggestion
            expect(
                act(() => {
                   result.current.handleClick();
                }),
            ).rejects.toThrow("Venmo session not available");
```

I'm fairly certain `async` / `await` isn't needed here. My tests didn't end up needing it. Though, I am a bit confused by the sparse documentation on `act()`. In this example, `async` / `await` _is_ used, but not the inner `await`; i.e. in this case, remove the `await` in `await result.current.handleClick()`.

https://legacy.reactjs.org/docs/testing-recipes.html#data-fetching

Reply:
Hmm for some reason my tests seem to fail if I don't use async / await. Using async / await makes sense to me since handleClick is an async function 🤔